### PR TITLE
[Azure Search] Add changelog for the past few SDK versions and couple of API versions

### PR DESCRIPTION
**Pull request by arv100kri**

### CODE REVIEW

**Comment by arv100kri** [comment]
Note: This is a draft PR because it needs to be serialized behind https://github.com/Azure/azure-sdk-for-net/pull/7265

For all intents and purposes, this is complete and can be reviewed.

File: sdk/search/Microsoft.Azure.Search/src/Microsoft.Azure.Search.csproj
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/src/Microsoft.Azure.Search.csproj]: Assuming the ChangeLog is ever growing and never pruned then having a link to master could work, assuming the location of the file doesn't move over time, but it might be better to use a commit sha or tag instead of master in your link.

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: @scbedd can confirm but I believe the version number needs to be under a heading 1 section (i.e. only 1 `#`) to be properly detected.

File: sdk/search/Microsoft.Azure.Search/src/Microsoft.Azure.Search.csproj
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/src/Microsoft.Azure.Search.csproj]: That's a reasonable point - to keep with the spirit, let me create another aka.ms link and point it to the master... that way, any changes to the location of the changelog wouldn't require a change to the csproj itself and it'll be similar to the SDK upgrade guide

Comment (weshaggard) [reply sdk/search/Microsoft.Azure.Search/src/Microsoft.Azure.Search.csproj]: While you could use an aka.ms link I'm not sure if you want to maintain that just for this. Even in that case it would always be pointing to the latest changelog which may or may not be what you want. If you want the change log to represent exactly what the source was when you shipped then you could use the sha.

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: This would read a bit better if you mention that it's the REST API version. Also, the plurality of API version in the second sentence is unclear. It should be one of:

> REST API versions targeted by...

or

> The REST API version targeted by...

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: They are, period. We don't put preview features in GA SDKs.

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: quotes -> quotas?

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: web api -> Web API
on creating skillset -> on creating a skillset

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: @arv100kri Thanks for digging through the history to create this!

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: FacetRestuls -> FacetResults

File: sdk/search/Microsoft.Azure.Search/changelog.md
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/changelog.md]: API version -> REST API versions

or

API version -> The REST API version

**Review by bitsforthought — CHANGES_REQUESTED** [review]

**Review by bitsforthought — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Comment by arv100kri** [comment]
> Note: This is a draft PR because it needs to be serialized behind #7265
> 
> For all intents and purposes, this is complete and can be reviewed.

This is now ready to be checked in. I'll merge this once #7265 is merged, which should be imminent

**Pull request merged**
